Framework/CurveFitting/src/Functions/ProcessBackground.cpp

```diff
@@ -366,11 +366,10 @@ void ProcessBackground::addRegion() {
   }
 
   // Check
-  for (auto it = vx.begin() + 1; it != vx.end(); ++it) {
-    if (*it <= *it - 1) {
-      g_log.error() << "The vector X with value inserted is not ordered incrementally\n";
-      throw std::runtime_error("Build new vector error!");
-    }
+  const auto it = std::adjacent_find(vx.cbegin(), vx.cend(), std::greater_equal<double>());
```

Comment:
Just want to check this is intended, it uses `std::greater_equal`, but the previous code uses `<=`?

Reply:
Yes, this is a bit confusing. In the original, it is checking element `i` is never less than or equal to element `i - 1` (the previous element). So essentially, checking that the vector is strictly (no repeating values) in ordered. You'd think you could just replace this with `std::is_ordered` but that will accept repeating values (`std::is_ordered({1, 1, 2, 3}) == True`).

`std::adjacent_find` searches for a pair `i` and `i + 1` (the next element) which matches a certain condition, in this case `i >= i+1`. If the algorithm finds a pair where this is true, we can also conclude the vector is not strictly ordered, so the two implementations should be equivalent. Maybe it needs a comment.

Reply:
Thanks for explaining!